### PR TITLE
Feature/kl 382

### PR DESCRIPTION
**Pull request by johnlac-21**
## Issue

[Fixed warnings and updated the project for Scala 2.13.5](https://github.com/knoldus/leaderboard/issues/382)

### Issue Type

* [ ] bug
* [ ] task
* [x] improvement
* [ ] story
* [ ] epic

## Changes

* Fixed all scapegoat warnings
* Fixed all scapegoat deprecations
* Added files so project can be run locally
* Upgraded old dependencies
* Fixed failed test cases

### Additional Details

...

## Integration Tests

Do ITs exist that will test these changes?

* [ ] no
* [x] yes
* [ ] this PR adds new ITs

## Postman

Do your changes include new or modified API calls?

* [x] no
* [ ] yes

If yes, has the Postman collection been updated?

* [ ] no
* [ ] yes
* [ ] not applicable

## Documentation

Do your changes require a change to the documentation?

* [ ] no
* [x] yes - internal
* [ ] yes - external

### CODE REVIEW

File: app/src/main/scala/com/knoldus/leader_board/business/KnolderRankImpl.scala [hunk 1]
```diff
@@ -14,30 +14,30 @@ class KnolderRankImpl extends KnolderRank with LazyLogging {
     val reputationPerKnolder = Vector.empty[GetReputation]
 
     @scala.annotation.tailrec
-    def getReputation(scorePerKnolder: List[GetScore], reputationPerKnolder: Vector[GetReputation], index: Int):
+    def getReputation(scrPerKnolder: List[GetScore], repPerKnolder: Vector[GetReputation], index: Int):
```
Comment (sonumehrotra) [inline app/src/main/scala/com/knoldus/leader_board/business/KnolderRankImpl.scala:17]: Why this change? Full names are contextually more sound.

File: app/src/main/scala/com/knoldus/leader_board/business/KnolderRankImpl.scala [hunk 2]
```diff
@@ -59,3 +59,5 @@ class KnolderRankImpl extends KnolderRank with LazyLogging {
     }
   }
 }
+
+
```
Comment (sonumehrotra) [inline app/src/main/scala/com/knoldus/leader_board/business/KnolderRankImpl.scala:63]: Extra line at EOF.

File: app/src/main/scala/com/knoldus/leader_board/business/KnolderRankImpl.scala (hunk 1 above)
Comment (johnlac-21) [reply app/src/main/scala/com/knoldus/leader_board/business/KnolderRankImpl.scala:17]: scapegoat was giving a warning about variable shadwoing

**Comment by flowpiyush** [comment]
guys can you also add the screen shot for - 
CPD
scalastyle
scapegoat
and test cases

@sara-naeem-cs @johnlac-21 @aidenknoldus

**Review by flowpiyush — APPROVED** [review]

**Comment by flowpiyush** [comment]
@sonumehrotra can we get approval from you and we can merge it.
